lib/experimental/navigation-theme-opt-in.php

```diff
@@ -191,36 +191,33 @@ function gutenberg_convert_menu_items_to_blocks(
 	$blocks = array();
 
 	foreach ( $menu_items as $menu_item ) {
-		if ( 'block' === $menu_item->type ) {
-			$parsed_blocks = parse_blocks( $menu_item->content );
-
-			if ( count( $parsed_blocks ) ) {
-				$block = $parsed_blocks[0];
-			} else {
-				$block = array(
-					'blockName' => 'core/freeform',
-					'attrs'     => array(
-						'originalContent' => $menu_item->content,
-					),
-				);
-			}
+		$menu_item_id = (int) $menu_item->ID;
+
+		$has_children = isset( $menu_items_by_parent_id[ $menu_item_id ] );
+
+		if ( $has_children ) {
+			$submenu_items = $menu_items_by_parent_id[ $menu_item_id ];
+			$block_name = 'core/navigation-submenu';
 		} else {
-			$block = array(
-				'blockName' => 'core/navigation-link',
-				'attrs'     => array(
-					'label' => $menu_item->title,
-					'url'   => $menu_item->url,
-				),
-			);
+			$submenu_items = array();
+			$block_name = 'core/navigation-link';
 		}
 
-		$block['innerBlocks'] = gutenberg_convert_menu_items_to_blocks(
-			isset( $menu_items_by_parent_id[ $menu_item->ID ] )
-					? $menu_items_by_parent_id[ $menu_item->ID ]
-					: array(),
-			$menu_items_by_parent_id
+		$block = array(
+			'blockName' => $block_name,
+			'attrs'     => array(
+				'label' => $menu_item->title,
+				'url'   => $menu_item->url,
+			),
 		);
-
+		$block_content = gutenberg_convert_menu_items_to_blocks(
```

Comment:
I don't think we can use this? This is in the experimental part of the defunct navigation editor, I think we have `block_core_navigation_parse_blocks_from_menu_items` for this use which is already used for `__unstableLocation``

Reply:
Updated to use this function

Reply:
We should then undo these changes?